src/formatting/cargo_toml.rs

```diff
@@ -242,8 +258,12 @@ impl VisitMut for KeyValue {
             // will remove decors and set the key to the bare key
             key.fmt();
         } else {
-            // add a space after the key
-            key.decor_mut().set_suffix(" ");
+            if value.is_value() {
+                // add a space after the key
+                key.decor_mut().set_suffix(" ");
+            } else {
+                key.decor_mut().set_suffix("");
+            }
         }
```

Comment:
It seems this fixed a different issue. I can merge this if you want to submit this separately. Ideally also add a test.

Reply:
Ah, apologies, I accidentally pushed this unrelated commit to this PR branch.